src/Workspaces/Remote/Core/Shared/SimpleAssetSource.cs

```diff
@@ -39,7 +39,7 @@ internal sealed class SimpleAssetSource : AssetSource
                 }
                 else
                 {
-                    Contract.Fail($"Unable to find asset for {checksum}");
+                    throw Contract.Unreachable;
```

Comment:
```suggestion
                    throw Contract.UnexpectedValue(checksum);
```

So we don't lose it from the message?